src/installer/scripts.py

```diff
@@ -126,9 +126,9 @@ def _get_launcher_data(self, kind: "LauncherKind") -> Optional[bytes]:
         key = (self.section, kind)
         try:
             name = _ALLOWED_LAUNCHERS[key]
-        except KeyError:
+        except KeyError as exc:
             error = f"{key!r} not in {sorted(_ALLOWED_LAUNCHERS)!r}"
-            raise InvalidScript(error)
+            raise InvalidScript(error) from exc
```

Comment:
```suggestion
            raise InvalidScript(error) from None
```

We shouldn't expose the `KeyError` to callers.